libraries/authoritative_helpers.rb

```diff
@@ -17,41 +17,74 @@
 # limitations under the License.
 #
 
-def default_authoritative_config_directory
-  case node['platform_family']
-  when 'debian'
-    '/etc/powerdns'
-  when 'rhel'
-    '/etc/pdns'
-  end
-end
+module PdnsAuthoritativeResource
+  module Helpers
+    def systemd_name(name = nil)
+      "pdns@#{name}"
+    end
 
-def default_authoritative_run_user
-  'pdns'
-end
+    def sysvinit_name(name = nil)
+      "pdns_authoritative-#{name}"
+    end
+
+    BACKENDS_PKG = Mash.new(
+        geo: 'pdns-backend-geo',
+        ldap: 'pdns-backend-ldap',
+        lua: 'pdns-backend-lua',
+        mysql: 'pdns-backend-mysql',
+        mydns: 'pdns-backend-mydns',
+        pipe: 'pdns-backend-pipe',
+        remote: 'pdns-backend-remote',
+        geoip: {
+          debian: 'pdns-backend-geoip',
+          rhel: nil,
+        },
+        odbc: {
+          debian: 'pdns-backend-odbc',
+          rhel: nil,
+        },
+        opendbx: {
+          debian: 'pdns-backend-opendbx',
+          rhel: nil,
+        },
+        tinydns: {
+          debian: 'pdns-backend-tinydns',
+          rhel: nil,
+        },
+        postgresql:{
+          debian: 'pdns-backend-pgsql',
+          rhel: 'pdns-backend-postgresql',
+        },
+        sqlite: {
+          debian: 'pdns-backend-sqlite3',
+          rhel: 'pdns-backend-sqlite',
+        },
+    ) unless defined? PdnsAuthoritativeResource::Helper::BACKENDS_PKG
 
-def backend_package_per_platform
-  return 'pdns-backend-geo'      if node['platform_family'] == 'debian' && new_resource.instance_name == 'geo'
-  return 'pdns-backend-ldap'     if node['platform_family'] == 'debian' && new_resource.instance_name == 'ldap'
-  return 'pdns-backend-mysql '   if node['platform_family'] == 'debian' && new_resource.instance_name == 'mysql'
-  return 'pdns-backend-pgsql'    if node['platform_family'] == 'debian' && new_resource.instance_name == 'postgresql'
-  return 'pdns-backend-pipe'     if node['platform_family'] == 'debian' && new_resource.instance_name == 'pipe'
-  return 'pdns-backend-sqlite3'  if node['platform_family'] == 'debian' && new_resource.instance_name == 'sqlite'
-  return 'pdns-backend-geoip'    if node['platform_family'] == 'debian' && new_resource.instance_name == 'geoip'
-  return 'pdns-backend-lua'      if node['platform_family'] == 'debian' && new_resource.instance_name == 'lua'
-  return 'pdns-backend-mydns'    if node['platform_family'] == 'debian' && new_resource.instance_name == 'mydns'
-  return 'pdns-backend-odbc'     if node['platform_family'] == 'debian' && new_resource.instance_name == 'odbc'
-  return 'pdns-backend-opendbx'  if node['platform_family'] == 'debian' && new_resource.instance_name == 'opendbx'
-  return 'pdns-backend-remote'   if node['platform_family'] == 'debian' && new_resource.instance_name == 'remote'
-  return 'pdns-backend-tinydns'  if node['platform_family'] == 'debian' && new_resource.instance_name == 'tinydns'
+    def backend_package_per_platform(name = nil)
```

Comment:
This was also inspired from pattern we also enforce in Criteo (https://github.com/criteo-cookbooks/wsus-client/blob/master/libraries/helper.rb#L3). This allow people to monkey patch the code in case we are not able to publish the code as fast as other contributors would like and we tend to think just having a Hash map is easier to maintain, does that make sense?